module/VuFind/src/VuFind/Search/Blender/Params.php

```diff
@@ -0,0 +1,655 @@
+                $result->add('fq', "-blender_backend:\"$backendId\"");
+            }
+        }
+        foreach ($this->searchParams as $params) {
+            $backendId = $params->getSearchClassId();
+            if (!is_callable([$params, 'getBackendParameters'])) {
+                throw new \Exception(
+                    "Backend $backendId missing support for getBackendParameters"
+                );
+            }
+
+            // Clone params so that adding any default filters does not affect the
+            // original instance:
+            $params = clone $params;
+            $this->addDefaultFilters($params, $backendId);
+
+            $result->set(
+                "query_$backendId",
+                $params->getQuery()
+            );
+            $result->set(
+                "params_$backendId",
+                $params->getBackendParameters()
+            );
+        }
+        return $result;
+    }
+
+    /**
+     * Add default filters to the given params
+     *
+     * @param BaseParams $params    Params
+     * @param string     $backendId Backend ID
+     *
+     * @return void
+     */
+    protected function addDefaultFilters(BaseParams $params, string $backendId): void
+    {
+        foreach ($this->mappings['Facets']['Fields'] ?? [] as $fieldConfig) {
+            $mappings = $fieldConfig['Mappings'][$backendId] ?? [];
+            $defaultValue = $mappings['DefaultValue'] ?? null;
+            if (null !== $defaultValue) {
+                $translatedField = $mappings['Field'];
+                $filterList = $params->getFilterList();
+                $found = false;
+                foreach ($filterList as $filters) {
+                    foreach ($filters as $filter) {
+                        if ($filter['field'] === $translatedField) {
+                            $found = true;
+                            break;
+                        }
+                    }
+                }
+                if (!$found) {
+                    $params->addFilter("$translatedField:$defaultValue");
+                }
+            }
+        }
+    }
+
+    /**
+     * Proxy a method call to parent class and all backend params classes
+     *
+     * @param string $method Method
+     * @param array  $params Method parameters
+     *
+     * @return mixed
+     */
+    protected function proxyMethod(string $method, array $params)
+    {
+        $result = call_user_func_array(['parent', $method], $params);
+        foreach ($this->searchParams as $searchParams) {
+            $result = call_user_func_array([$searchParams, $method], $params);
+        }
+        return $result;
+    }
+
+    /**
+     * Translate a facet
+     *
+     * @param string $field     Facet field
+     * @param string $backendId Backend ID
+     *
+     * @return string
+     */
+    protected function translateFacet(string $field, string $backendId): string
+    {
+        $prefix = '';
+        if (substr($field, 0, 1) === '~') {
```

Comment:
It would be useful if we had a clearer distinction between a facet and a filter, but basicly I consider a facet here a facet field configuration while a filter is an active filter. A filter actually doesn't need to have anything to do with active facets, though it's usually enabled via a facet entry. But back to the actual issue, I'm not quite sure if it needs both or none, but the explanation for the current situation probably is that the tests exercise that.

This all boils down to the question: If you call `removeAllFilters('something')`, should it remove all OR and NOT filters as well (I think it should, but currently doesn't), or should you issue a `removeAllFilters('~something')` etc. as well? Currently it only does the exact version, but I'm not sure if it makes much sense. In addition, with Blender a single filter value may map to multiple values, so `foo:bar` may be translated to `~foo:baf` OR `~foo:baz` for a backend. If the caller then tries `removeAllFilters('foo')`, these OR filters don't get removed.

Reply:
I would also expect that removeAllFilters should do what it says and remove OR/NOT as well as "plain" filters, and my guess would be that, if it does not, that's an oversight. However, it's also entirely possible that I'm forgetting something. Have you done any analysis of where/how in the code removeAllFilters is actually getting called? If changing the behavior is justified, maybe that work should be done in a separate PR to make it easier to keep track of.

Reply:
It occurs to me that if we wanted to do this as cautiously as possible, we could add a second parameter to removeAllFilters to specify which operator(s) to remove. If the current behavior is clearly wrong, I wouldn't want to make things more complicated, but if there is a use case for targeting a specific operator, that might be a better/more explicit way to do it.

Reply:
I can't really see it being used anywhere else than `getPartialFieldFacets`, and there it seems that the calling code would expect it to remove all filters.

Reply:
Agreed. Maybe we should consider this a bug fix against dev so we can correct the newly-added tests to behave as expected. Do you have time for a PR, or should I put this on my list?